src/components/messages/MessageBox/index.tsx

```diff
@@ -13,6 +14,14 @@ const MessageBox: React.FC<MessageBoxProps> = ({ topic, authorAccount }) => {
   const [messageText, setMessageText] = useState('')
   const { chatClientProxy } = useContext(W3iContext)
 
+  const isDisabled = useMemo(
+    () =>
+      (topic.includes('invite:')
+        ? (topic.split(':')[1] as ChatClientTypes.SentInvite['status'])
+        : 'approved') !== 'approved',
+    [topic]
```

Comment:
You can use the `w3iProxy` and retrieve `sentInvites` from the context and query them to find the actual status

Reply:
Yep, gotcha!